### ISSUE TITLE
.NET Tools update still not work on non-English systems.

### ISSUE THREAD

**Issue by Jisu-Woniu** [topgrade-rs/topgrade#397]
## What did you expect to happen?

.NET tools upgraded successfully after using `topgrade` command.

## What actually happened?

.NET upgrade is omitted during topgrade process.

## Additional Details
- Which operating system or Linux distribution are you using? Arch Linux
- How did you install Topgrade? Through AUR
- Which version are you running? 10.3.3

![图片](https://user-images.githubusercontent.com/31986081/228786601-abde475e-54b9-41fa-99e8-2f776f43f8ef.png)

## Reason for this issue

My computer is running with Chinese Simplified locale (`LANG=zh_CN.UTF-8`), and the output of `dotnet tool list -g` does not start with `Package Id`. (Line 593 below)

https://github.com/topgrade-rs/topgrade/blob/565aa405be6d123ae87cf9cc71f04b01bb66d026/src/steps/generic.rs#L578-L595

![图片](https://user-images.githubusercontent.com/31986081/228788148-2c79a76f-a04c-45ae-a270-6e0179c1898f.png)

```
$ dotnet tool list -g
包 ID           版本          命令
-------------------------------------------
csharpier      0.23.0      dotnet-csharpier
fantomas       5.2.4       fantomas
```

**Comment by guihkx** [comment]
> You should use a locale insensitive way to detect whether the output contains packages.
> 
> For example, just check whether the output contains a line of `---...-`

It would probably make more sense to invoke external commands with `LC_ALL=C` set instead.

**Comment by SteveLauC** [comment]
Hi @Jisu-Woniu, would you like to try:
```shell
$ LC_ALL=C topgrade --verbose --only dotnet
```

and see if this walkaround works

**Comment by Jisu-Woniu** [comment]
That did work, but it should be done internally, and Windows machines may not have the `LC_ALL` envvar.

**Comment by SteveLauC** [comment]
> That did work, but it should be done internally, 

Indeed

> and Windows machines may not have the `LC_ALL` envvar.

I did a quick search yesterday and found that windows does have some equivalent stuff, but haven't figured out how I can access it in Rust.

Will dig it deeper later and hopefully get this issue fixed:)

**Comment by Jisu-Woniu** [comment]
I think that's easy, just remove the test in 

https://github.com/topgrade-rs/topgrade/blob/c20a300eeaeb7a959a8d805bbc83a726e5da6624/src/steps/generic.rs#L631

Seems it is from this commit: https://github.com/topgrade-rs/topgrade/commit/7f03f33d6b320092fa59164c63f5b9160f3074ff

Gonna check it in a docker container with only .NET runtime installed.

**Comment by Jisu-Woniu** [comment]
I think the check below is sufficient to tell SDK and runtime apart, and the test above is redundant.

https://github.com/topgrade-rs/topgrade/commit/1b5a9ae67f038c673acf90569dc620fcf2929dd7

**Comment by Jisu-Woniu** [comment]
Here is the output of `dotnet tool`

![image](https://github.com/topgrade-rs/topgrade/assets/31986081/9bdcdef4-71e8-4b64-a700-e5f4198a23df)

**Comment by SteveLauC** [comment]
With ONLY `runtime` installed, `dotnet tool list -g` gives:

```shell
$ dotnet tool list -g
The command could not be loaded, possibly because:
  * You intended to execute a .NET application:
      The application 'tool' does not exist.
  * You intended to execute a .NET SDK command:
      No .NET SDKs were found.

Download a .NET SDK:
https://aka.ms/dotnet/download

Learn about SDK resolution:
https://aka.ms/dotnet/sdk-not-found
```

With ONLY `SDK` and no package installed, `dotnet tool list -g` gives:

```shell
$ dotnet tool list -g
Package Id      Version      Commands
-------------------------------------
```

> Tested with `.NET 7.0` and `Fedora 37`

-----
To me, it seems the [check for `Pakcage Id`](https://github.com/topgrade-rs/topgrade/blob/c20a300eeaeb7a959a8d805bbc83a726e5da6624/src/steps/generic.rs#L631-L633) is indeed redundant as **it will always be printed**, is this right? @Jisu-Woniu

**Comment by Jisu-Woniu** [comment]
No, it isn't the case in non-English systems.

**Comment by SteveLauC** [comment]
> No, it isn't the case in non-English systems.
> 
> Refer to the initial post.

Sorry for being unclear, let me rephrase my statement:
```
Package Id      Version      Commands
-------------------------------------
```
This header (**in English or whatever language**) will be always printed regardless of if there is a package installed or not, is this right?

```shell
$ LANG=zh_CN.UTF-8 dotnet tool list -g
包 ID      版本      命令
--------------------
```

If so, then we can safely remove it.

**Comment by Jisu-Woniu** [comment]
Yes
